pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,39 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// if we have both an ICT defined and an empty Template->PodSpec->Container->Image field, we are going
```

Comment:
Yeah, the span of mutation had originally occurred to me as well, but since the parameter into `ValidateDeploymentConfigSpec` is NOT a pointer, my understanding was that golang would pass in a copy of the param to the method.

Looking again though, while the `DeploymentConfigSpec` passed in is not a pointer, the underlying `Template` field is a pointer.  Perhaps then even with `DeploymentConfigSpec` parameter being a "copy", the Template pointer field with be the same address in both `DeploymentConfigSpec` copies ... as I type, I'm thinking yes, but confirmation / correction of my train of thought here would be great.

I'll start working on the changes in the interim.